pygments/lexers/elpi.py

```diff
@@ -0,0 +1,143 @@
+    idcharstarns_re = r"({}+|(?=\.[a-z])\.{}+)".format(idchar_re,idchar_re)
+    symbchar_re = r"({}|{}|{}|{}|:)".format(lcase_re, ucase_re, digit_re, schar_re)
+    constant_re = r"({}{}*|{}{}*|{}{}*|_{}+)".format(ucase_re, idchar_re, lcase_re, idcharstarns_re,schar2_re, symbchar_re,idchar_re)
+    symbol_re=r"(,|<=>|->|:-|;|\?-|->|&|=>|\bas\b|\buvar\b|<|=<|=|==|>=|>|\bi<|\bi=<|\bi>=|\bi>|\bis\b|\br<|\br=<|\br>=|\br>|\bs<|\bs=<|\bs>=|\bs>|@|::|\[\]|`->|`:|`:=|\^|-|\+|\bi-|\bi\+|r-|r\+|/|\*|\bdiv\b|\bi\*|\bmod\b|\br\*|~|\bi~|\br~)"
+    escape_re=r"\(({}|{})\)".format(constant_re,symbol_re)
+    const_sym_re = r"({}|{}|{})".format(constant_re,symbol_re,escape_re)
+
+    tokens = {
+        'root': [ include('elpi') ],
+
+        'elpi': [
+
+            include('_elpi-comment'),
+
+            (r"(:before|:after|:if|:name)(\s*)(\")",bygroups(Keyword.Mode,Text,String.Double),'elpi-string'),
+            (r"(:index)(\s*\()",bygroups(Keyword.Mode,Text),'elpi-indexing-expr'),
+            (r"\b(external pred|pred)(\s+)({})".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-pred-item'),
+            (r"\b(external type|type)(\s+)(({}(,\s*)?)+)".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-type'),
+            (r"\b(kind)(\s+)(({}|,)+)".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-type'),
+            (r"\b(typeabbrev)(\s+)({})".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-type'),
+            (r"\b(accumulate)(\s+)(\")",bygroups(Keyword.Declaration,Text,String.Double),'elpi-string'),
+            (r"\b(accumulate|namespace|local)(\s+)({})".format(constant_re),bygroups(Keyword.Declaration,Text,Text)),
+            (r"\b(shorten)(\s+)({}\.)".format(constant_re),bygroups(Keyword.Declaration,Text,Text)),
+            (r"\b(pi|sigma)(\s+)([a-zA-Z][A-Za-z0-9_ ]*)(\\)",bygroups(Keyword.Declaration,Text,Name.Variable,Text)),
+            (r"\b(constraint)(\s+)(({}(\s+)?)+)".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-chr-rule-start'),
+
+            (r"(?=[A-Z_]){}".format(constant_re),Name.Variable),
+            (r"(?=[a-z_]){}\\".format(constant_re),Name.Variable),
+            (r"_",Name.Variable),
+            (r"({}|!|=>|;)".format(symbol_re),Keyword.Declaration),
+            (constant_re,Text),
+            (r"\[|\]|\||=>",Keyword.Declaration),
+            (r'"', String.Double, 'elpi-string'),
+            (r'`', String.Double, 'elpi-btick'),
+            (r'\'', String.Double, 'elpi-tick'),
+            (r'\{[^\{]', Text, 'elpi-spill'),
+            (r"\(",Text,'elpi-in-parens'),
+            (r'\d[\d_]*', Number.Integer),
+            (r'-?\d[\d_]*(.[\d_]*)?([eE][+\-]?\d[\d_]*)', Number.Float),
+            (r"[\+\*\-/\^\.]", Operator),
+        ],
+        '_elpi-comment': [
+            (r'%[^\n]*\n',Comment),
+            (r'/\*',Comment,'elpi-multiline-comment'),
+            (r"\s+",Text),
```

Comment:
Same as above, `\s` should be matched `Text.Whitespace` (not going to highlight the remaining cases.)

Reply:
done